src/main/webapp/management-console.js

```diff
@@ -304,38 +305,60 @@ angular.module('managementConsole', [
             .when('/', '/login')
             .when('', '/login')
             .otherwise('/error404');
+    }])
+
+  .config(['localStorageServiceProvider', function(localStorageServiceProvider) {
+    localStorageServiceProvider.setPrefix('infinispan-management-console');
   }])
+
 /**
  * Safe apply method. It should be used when normal $scope.$apply happens to execute
  * during digest cycle which causes an error.
  * Use it just like normal apply: $scope.safeApply(myFunc).
  */
   .run(['$rootScope', '$timeout', function ($rootScope, $timeout) {
-      $rootScope.safeApply = function (f) {
-        var scope = this;
-        $timeout(function () {
-          scope.$apply(f);
-        });
-      };
-      $rootScope.page = {htmlClass: ''};
-    }]).run(['$rootScope', '$location', 'modelController', function ($rootScope, $location, modelController) {
+    $rootScope.safeApply = function (f) {
+      var scope = this;
+      $timeout(function () {
+        scope.$apply(f);
+      });
+    };
+    $rootScope.page = {htmlClass: ''};
+  }])
 
-      $rootScope.$on('$locationChangeStart', function () {
+  .run([
+    '$rootScope', 'modelController', '$urlRouter', '$state',
+    function ($rootScope, modelController, $urlRouter, $state) {
+      $rootScope.$on('$stateChangeStart', function (event, toState) {
         // redirect to login page if not logged in
-        if ($location.path() !== '/login' && !modelController.isAuthenticated()) {
-          $location.path('/login');
+        if (toState.name !== 'login' && !modelController.isAuthenticated()) {
+          event.preventDefault();
+          modelController.login().then(function () {
+            // TODO: Do I have to do this refreshing here? Why?
```

Comment:
It is ok actually - there are two ways a user can login, either by this part of code which executes on any state change (and also on (re)load of application), or through login controller.
If his credentials are stored and they are correct, login will normally happen here. If he logouts and credentials are erased (or he has never logged in before) this part of code will reject him and send him to login controller, where he will be able to do a login by inputing his credentials.

Model refresh will not happen twice because once that login executes successfuly (either here or in login controller) that is remembered in modelController (isAuthenticated() method) and since we check for that method here, it can not happen again!

I asked if we need this refresh here because I did not know what it is for. But if we needed it in login controller (as you said), then we need it here also.